### ISSUE TITLE
Next15 Support | Cannot convert undefined or null to object at Function.entries

### ISSUE THREAD

**Issue by nagarajpandith** [aidansunbury/trpc-ui#53]
### Describe the bug
Looks like renderTrpcPanel function in trpc-ui does not support the new Route Handlers in Next.js 15, resulting in an error when attempting to use it. Specifically, the appRouter variable is undefined or null when invoked within a GET handler.

### To Reproduce
1. Set up a Next.js 15 project with the trpc-ui package
2. Attempt to create a GET handler in the /app/api/panel/route.ts file to render the tRPC panel:
```ts
import { renderTrpcPanel } from "trpc-ui";
import { NextResponse } from "next/server";
import { appRouter } from "@/lib/server/routers/_app";

export async function GET() {
  const html = renderTrpcPanel(appRouter, {
    url: "http://localhost:3000/api/trpc",
    transformer: "superjson",
  });

  return new NextResponse(html, { status: 200, headers: { "Content-Type": "text/html" } });
}
```
3. Run the application and navigate to /api/panel
4. Observe the error: appRouter is undefined or null.

**trpc-ui version:** v1.0.10

**trpc [adapter](https://trpc.io/docs/server/adapters) used:** @trpc/server/adapters/next

**Are you using superjson:** yes

### Expected behavior
The renderTrpcPanel function should work seamlessly with the new Route Handlers in Next.js 15, allowing developers to render the tRPC UI without errors.

### Screenshots
<img width="439" alt="Screenshot 2024-12-29 at 10 36 51 PM" src="https://github.com/user-attachments/assets/55271fd5-7414-4187-b8ea-08c483c660b6" />

### Additional context
NA

👍: I have this issue
👎: This issue has already been fixed
🚀: I would like to work on / be assigned this issue (@aidansunbury to get assigned)

**Comment by nagarajpandith** [comment]
@aidansunbury i see you had self-assigned this 3 days back. any updates? thanks.

**Comment by aidansunbury** [comment]
Sorry! Have been busy over the new year, I'll try and get it done tomorrow

**Comment by aidansunbury** [comment]
@nagarajpandith @jeanpfs @svengau are you sure that you are importing the appRouter from the correct place? 

I tried the provided code snippet in a repo using nextjs 15.1.3 and everything seemed to be working fine. Are you able to provide a repository reproducing the issue you are encountering? 

My repo: https://github.com/aidansunbury/next-15-trpc-ui

**Comment by nagarajpandith** [comment]
@aidansunbury sorry that i didn't mention about kirimase. the trpc setup was done using [kirimase](https://kirimase.dev/getting-started). here's a demo repo to reproduce this bug: https://github.com/nagarajpandith/trpc-ui-bug (it's a pvt repo, i have invited you).

```bash
git clone https://github.com/nagarajpandith/trpc-ui-bug
cd trpc-ui-bug
pnpm i
pnpm dev
```

if you have any dummy postgres db, you can fill in the env and run `pnpm db:generate`, but this bug is not really related to that.

**Comment by aidansunbury** [comment]
@nagarajpandith The issue doesn't have anything to do with nextjs 15, the problem is that you are using trpc v10 and not trpc v11. `trpc-ui` currently isn't backwards compatible with v10, but you can still use the only `trpc-panel` package. Just `pnpm install trpc-panel` and replace your current import to instead be 

```ts
import { renderTrpcPanel } from "trpc-panel";
```

I will open an issue about potentially supporting v10 backwards compatibility.

**Comment by nagarajpandith** [comment]
awesome, trpc-panel works well for me but i wanted to use trpc-ui as trpc-panel is now abandoned by the original author. anyways, until you introduce a backwards compatibility i will be using trpc-panel. this issue can now be closed.

**Comment by svengau** [comment]
Thanks for investigating @aidansunbury. 

For the record, It turns out my issue was related to the use of `discriminatedUnion` with `enum.extract(...)`. It's a valid zod schema, but it doesn't seem to be well-supported by trpc-ui.

Here is a sample schema:
```
const discriminatedFieldEnum = z.enum(["One", "Two"]); 
[...]
discriminatedUnionInput: t.procedure
      .input(
        z.object({
          aDiscriminatedUnion: z.discriminatedUnion("discriminatedField", [
            z.object({
              discriminatedField: discriminatedFieldEnum.extract(["One"]), // <-- this doesn't work 
              aFieldThatOnlyShowsWhenValueIsOne: z.string(),
            }),
            z.object({
              discriminatedField: z.literal("Two"),
              aFieldThatOnlyShowsWhenValueIsTwo: z.object({
                someTextFieldInAnObject: z.string(),
              }),
            }),
          ]),
        }),
      )
```

Easy workaround: simply replace `enum.extract(...)` by `z.literal(...)`

**Comment by aidansunbury** [comment]
> Thanks for investigating @aidansunbury.
> 
> For the record, It turns out my issue was related to the use of `discriminatedUnion` with `enum.extract(...)`. It's a valid zod schema, but it doesn't seem to be well-supported by trpc-ui.
> 
> Here is a sample schema:
> 
> ```
> const discriminatedFieldEnum = z.enum(["One", "Two"]); 
> [...]
> discriminatedUnionInput: t.procedure
>       .input(
>         z.object({
>           aDiscriminatedUnion: z.discriminatedUnion("discriminatedField", [
>             z.object({
>               discriminatedField: discriminatedFieldEnum.extract(["One"]), // <-- this doesn't work 
>               aFieldThatOnlyShowsWhenValueIsOne: z.string(),
>             }),
>             z.object({
>               discriminatedField: z.literal("Two"),
>               aFieldThatOnlyShowsWhenValueIsTwo: z.object({
>                 someTextFieldInAnObject: z.string(),
>               }),
>             }),
>           ]),
>         }),
>       )
> ```
> 
> Easy workaround: simply replace `enum.extract(...)` by `z.literal(...)`

Yes unfortunately the default behavior at the moment is to not render a procedure if the zod validator can't be parsed. At some point I will try and make it such that you can still use the json editor even if the zod validator can't be parsed.

Also enum.extract / exclude should be supported at some point. If it is something that you use frequently you should open an issue.
